frontend-cert/js-projects/hash-table/user-stories.md

```diff
@@ -0,0 +1,11 @@
+1. You should define a class named `HashTable` with a `collection` property initialized as an empty object inside its `constructor` method. The `collection` object should store key-value pairs based on the hashed value of the key.
+
+1. The `HashTable` class should have a static method `hash` and three instance methods, namely `add`, `remove`, and `lookup`.
```

Comment:
There are no guarantees that the static method will be mention in the lectures because it will be up to the lecture author. 

So to be on the safe side, I would change this to have them add another property to the constructor instead

Reply:
Do you mean like this?
```js
class HashTable {
    constructor() {
        this.collection = {};
        this.hash = function(string) {
            let hashed = 0;
            for (let i = 0; i < string.length; i++) {
                hashed += string.charCodeAt(i);
            }
            return hashed
        }
    }
    ...
}
```
Or just a regular instance method?

Reply:
Yeah, you can ask them to add another property.
Or you can ask them to add a regular instance method.
Either is fine.

The only thing you can't ask them is to add a static method because that won't be taught.